src/cli/platform/caching.rs

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Context;
+use std::fs::OpenOptions;
+use std::io::{Read, Write};
+use std::path::Path;
+
+use sha2::{Digest, Sha256};
+
+/// Fetch a URL and save the contents as the hash of the URL
+pub fn save_file(url: &str, dest: &Path) -> anyhow::Result<()> {
+    let response = ureq::get(url)
+        .call()
+        .context(format!("retrieving CRL {url} from server"))?
+        .into_reader();
+    let response: Vec<u8> = response.bytes().map(|d| d.unwrap()).collect();
+
+    let mut dest = dest.to_path_buf();
+    dest.push(hex::encode(Sha256::digest(url)));
+
+    let mut file = OpenOptions::new()
+        .create(true)
+        .write(true)
+        .open(&dest)
+        .context(format!(
+            "opening destination file {:?} for saving CRL {url}",
+            dest.display()
+        ))?;
+
+    file.write_all(&response).context(format!(
+        "writing CRL file {} to disk from {url}",
+        dest.display()
+    ))?;
+    Ok(())
+}
```

Comment:
```suggestion
pub fn save_file(url: &str, dest: &Path) -> anyhow::Result<()> {
    let mut response = ureq::get(url)
        .call()
        .context(format!("retrieving CRL {url} from server"))?
        .into_reader();

    let mut dest = dest.to_path_buf();
    dest.push(hex::encode(Sha256::digest(url)));

    let mut file = OpenOptions::new()
        .create(true)
        .write(true)
        .open(&dest)
        .context(format!(
            "opening destination file {:?} for saving CRL {url}",
            dest.display()
        ))?;

    std::io::copy(&mut response, &mut file).context(format!(
        "saving CRL {url} to destination file {:?}",
        dest.display()
    ))?;

    Ok(())
}
```

Reply:
Thank you for that. I don't understand what I did wrong when I tried this, but I was getting errors about `Box<...>` not implementing some needed trait. Works now.